src/router/routes/genericResourceRoute.ts

```diff
@@ -127,31 +128,56 @@ export default class GenericResourceRoute {
         }
 
         if (this.operations.includes('search-type')) {
+            const handleSearch = async (res: express.Response, resourceType: string, searchParamQuery: any) => {
+                const allowedResourceTypes = await this.authService.getAllowedResourceTypesForOperation({
+                    operation: 'search-type',
+                    userIdentity: res.locals.userIdentity,
+                });
+
+                const response = await this.handler.typeSearch(
+                    resourceType,
+                    searchParamQuery,
+                    allowedResourceTypes,
+                    res.locals.userIdentity,
+                );
+                const updatedSearchResponse = await this.authService.authorizeAndFilterReadResponse({
+                    operation: 'search-type',
+                    userIdentity: res.locals.userIdentity,
+                    readResponse: response,
+                });
+                return updatedSearchResponse;
+            };
             // SEARCH
             this.router.get(
                 '/',
                 RouteHelper.wrapAsync(async (req: express.Request, res: express.Response) => {
                     // Get the ResourceType looks like '/Patient'
                     const resourceType = req.baseUrl.substr(1);
                     const searchParamQuery = req.query;
+                    const updatedSearchResponse = await handleSearch(res, resourceType, searchParamQuery);
+                    res.send(updatedSearchResponse);
+                }),
+            );
+            this.router.post(
+                '/_search',
+                RouteHelper.wrapAsync(async (req: express.Request, res: express.Response) => {
+                    // Get the ResourceType looks like '/Patient'
+                    const resourceType = req.baseUrl.substr(1);
+                    const searchParamQuery = req.query;
+                    const { body } = req;
 
-                    const allowedResourceTypes = await this.authService.getAllowedResourceTypesForOperation({
-                        operation: 'search-type',
-                        userIdentity: res.locals.userIdentity,
-                    });
+                    if (!isEmpty(body)) {
+                        mergeWith(searchParamQuery, body, (valueOne, valueTwo) => {
+                            const result: string | any[] = [];
+                            if (!isEmpty(valueOne) && !isEmpty(valueTwo) && !(valueOne === valueTwo)) {
```

Comment:
There is the case when both GET and POST params are arrays:
```
parametersInQuery=active=true&active=false
parametersInBody=active=true&active=false
```

This code works in that case but it is misleading since it does `valueOne === valueTwo` and equality comparisons between arrays is always false.

I suggest to simply drop the `!(valueOne === valueTwo)` condition. Duplicated params are removed anyways by the Set

Reply:
If `!(valueOne === valueTwo)` is dropped, then the situation where `parametersInQuery=active=true` and `parametersInBody=active=true` would result in [true] instead of true.

Reply:
Right.

Still, let's not do `===` comparisons on arrays. We can be explicit about merging arrays OR unique values by replacing `!(valueOne === valueTwo)` with something like `(Array.isArray(valueOne) || Array.isArray(valueTwo) || valueOne !== valueTwo)`

btw, deduplicating params wasn't strictly necessary, although it doesn't hurt to have it here. Maybe that logic fits better in the Search component since it ends up being only an optimization on the resulting search queries.

Reply:
`(Array.isArray(valueOne) || Array.isArray(valueTwo) || valueOne !== valueTwo)` won't work for situation like `parametersInQuery=active=true` and `parametersInBody=gender=female`. It'll result in `active=[true, undefined]`, `gender=[undefined,female]`. We'll still need the block `(!isEmpty(valueOne) || !isEmpty(valueTwo)` to return undefined before it. 

I agree it's not ideal to use `===` on arrays, but the alternative is getting the conditions more complicated. I can add a comment here explaining what situation it covers. Does that work?

Reply:
sorry, I meant replacing the last part of the condition, not all of it. 
i.e.
```
!isEmpty(valueOne) &&
!isEmpty(valueTwo) &&
(Array.isArray(valueOne) || Array.isArray(valueTwo) || valueOne !== valueTwo)
```
yes, the conditions are more complex, but I think that they are easier to understand as a developer (95% of the time when you see an array `==` comparison it is a bug). It's kind of adding more code to make your intent clear instead of adding a comment to explain it.

Reply:
Okay, I'll update the condition to avoid `===` on arrays.